server/src/collection.rs

```diff
@@ -205,6 +205,29 @@ impl Collection {
         None
     }
 
+    pub async fn set_locked(&self, locked: bool) -> Result<(), ServiceError> {
+        let items = self.items.lock().await;
+
+        for item in items.iter() {
+            item.set_locked(true).await?;
+        }
+
+        let manager = self.manager.lock().await;
+
+        if !self.is_locked().await {
+            self.locked
+                .store(locked, std::sync::atomic::Ordering::Relaxed);
+            let signal_emitter = manager.signal_emitter(&self.path)?;
+            self.locked_changed(&signal_emitter).await?;
+        }
+
+        let service_path = OwnedObjectPath::try_from("/org/freedesktop/secrets").unwrap();
```

Comment:
You can get the path from oo7::dbus

Reply:
`OwnedObjectPath::try_from(oo7::dbus::api::Service::PATH.as_deref().unwrap()).unwrap()` has a few unwrap. Hope this is ok?